api/client/retl/sources.go

```diff
@@ -0,0 +1,108 @@
+package retl
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// CreateRetlSource creates a new RETL source
+func (r *RudderRETLStore) CreateRetlSource(ctx context.Context, source *RETLSourceCreateRequest) (*RETLSource, error) {
+	data, err := json.Marshal(source)
```

Comment:
I see in some cases (e.g create) we marshal the request directly, while in others (e.g update, put state) we populate a map from the request and marshal that instead. It's better if we have one approach for every case where we need to create a request body. Either serialize the request (relying on json tags) or directly have the client functions populate the bodies from the request arguments.